tests/template/test_ring_template.py

```diff
@@ -46,3 +50,51 @@ def test_ring_default(hdf5_ds_2, tmpdir_factory, lt_ctx):
         results,
         expected['intensity'].raw_data,
     )
+
+
+@pytest.mark.dist
+@pytest.mark.asyncio
+def test_ring_tcp_cluster(hdf5_ds_2, tmpdir_factory, lt_ctx):
+    datadir = tmpdir_factory.mktemp('template_tests')
+
+    conn = {
+        'connection': {
+            'type': 'TCP',
+            'address': 'tcp://scheduler:8786',
+        }
+    }
+    path = hdf5_ds_2.path
```

Comment:
The test is failing because of the path, is there any difference for providing dataset path when using cluster in the test?

Reply:
Yes: the workers (and the tests themselves) are started in their own containers, and as such don't have access to each others file systems.

Currently, the `data` subdirectory is bind mounted into each container at `/data/`, such that you can put a variety of test data sets in there. Maybe you can create a temporary directory in the `data` directory, and use that path for these tests? Then it is available on each worker container.

Reply:
[Here](https://github.com/LiberTEM/LiberTEM/blob/16011aee5fab65fc2a4094ddb4e1a203f5d51b3e/conftest.py#L38) the dataset is created in the `/data/` right?

Reply:
> the data subdirectory is bind mounted into each container at /data/

Is this [dataset](https://github.com/LiberTEM/LiberTEM/blob/16011aee5fab65fc2a4094ddb4e1a203f5d51b3e/tests/io/datasets/test_blo.py#L17) also created in the `/data/` ? `test_blo` is currently being skipped on my local machine.

Reply:
> [Here](https://github.com/LiberTEM/LiberTEM/blob/16011aee5fab65fc2a4094ddb4e1a203f5d51b3e/conftest.py#L38) the dataset is created in the `/data/` right?

No - it is created in a temporary directory, with `data` somewhere in the name. So most of the time this will end up in `/tmp/`

Reply:
> > the data subdirectory is bind mounted into each container at /data/
> 
> Is this [dataset](https://github.com/LiberTEM/LiberTEM/blob/16011aee5fab65fc2a4094ddb4e1a203f5d51b3e/tests/io/datasets/test_blo.py#L17) also created in the `/data/` ? `test_blo` is currently being skipped on my local machine.

To be clear, the `blo` tests (and most other dataset tests, other than raw and hdf5) require the user to put specific, not yet publicly available, datasets at specific paths. It's not really possible to generate the data for these formats on the fly, as we need bit-for-bit matching input data, which can be error-prone. This is a bit painful, but that's how it is currently.

So, for these formats, there is the shared `/data/` bind mount available on the worker processes.

> Maybe you can create a temporary directory in the data directory, and use that path for these tests? Then it is available on each worker container.

Another possibility is to create a dedicated shared volume for these, mount it at `/shared-tmp-data/` or similar in all containers, and create the temporary test files there. That could be a better idea, as it means the `/data`/ mount can be treated as read-only.